src/BuildScriptGeneratorCli/Commands/DetectCommand.cs

```diff
@@ -0,0 +1,123 @@
+            var logger = loggerFactory.CreateLogger<DetectCommand>();
+            var sourceRepo = new LocalSourceRepo(SourceDir, loggerFactory);
+            var ctx = new DetectorContext
+            {
+                SourceRepo = sourceRepo,
+            };
+
+            var detector = serviceProvider.GetRequiredService<IDetector>();
+            var detectedPlatforms = detector.GetAllDetectedPlatforms(ctx);
+            using (var timedEvent = logger.LogTimedEvent("DetectCommand"))
+            {
+                if (detectedPlatforms != null && detectedPlatforms.Any())
+                {
+                    if (OutputJson)
+                    {
+                        console.WriteLine("Detection result in Json format:");
+                        console.WriteLine(JsonFormatResult(detectedPlatforms));
+                    }
+                    else
+                    {
+                        console.WriteLine("Detection result:");
+                        console.WriteLine(ListFormatResult(detectedPlatforms));
+                    }
+
+                    return ProcessConstants.ExitSuccess;
+                }
+            }
+
+            return ProcessConstants.ExitFailure;
+        }
+
+        internal override bool IsValidInput(IServiceProvider serviceProvider, IConsole console)
+        {
+            var logger = serviceProvider.GetRequiredService<ILogger<DetectCommand>>();
+            SourceDir = string.IsNullOrEmpty(SourceDir) ? Directory.GetCurrentDirectory() : Path.GetFullPath(SourceDir);
+
+            if (!Directory.Exists(SourceDir))
+            {
+                logger?.LogError("Could not find the source directory.");
+                console.WriteErrorLine($"Could not find the source directory: '{SourceDir}'");
+
+                return false;
+            }
+
+            return true;
+        }
+
+        private string ListFormatResult(IEnumerable<PlatformDetectorResult> detectedPlatforms)
+        {
+            var result = new StringBuilder();
+
+            foreach (var detectedPlatform in detectedPlatforms)
+            {
+                var defs = new DefinitionListFormatter();
+
+                defs.AddDefinition("Platform", detectedPlatform.Platform);
+                defs.AddDefinition("Version", detectedPlatform.PlatformVersion ?? "N/A");
+
+                result.AppendLine(defs.ToString());
+            }
+
+            return result.ToString();
+        }
+
+        private string JsonFormatResult(IEnumerable<PlatformDetectorResult> detectedPlatforms)
+        {
+            var detectionResult = new Dictionary<string, string>
+            {
+                ["App_Path"] = SourceDir,
+            };
+            var platformData = new List<Dictionary<string, string>>();
+            foreach (var detectedPlatform in detectedPlatforms)
+            {
+                Dictionary<string, string> data = new Dictionary<string, string>
```

Comment:
Since a detector result could have more information than just the name and version, we should not be doing this. To make it simpler we could just serialize the list of platform detector result itself. In my opinion things like `App_Path` are not interested to be put into the output.

Reply:
Good catch, I forgot that the fact that detector results could have more properties, will fix this